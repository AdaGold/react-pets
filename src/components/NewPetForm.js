import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NewPetForm extends Component {
  static propTypes = {
    addPetCallback: PropTypes.func.isRequired
  };

  constructor() {
    super();

    this.state = {
      name: '',
      breed: '',
      age: '',
      about: ''
    };
  }

  onInputChange = (event) => {
    const newState = {};

    newState[event.target.name] = event.target.value;

    this.setState(newState);
  }

  resetState = () => {
    this.setState({
      name: '',
      breed: '',
      age: '',
      about: ''
    });
  }

  onFormSubmit = (event) => {
    event.preventDefault();
    console.log(this.state);
    this.props.addPetCallback(
      this.state);

    this.resetState();
  }

  render() {

    return (
      <form
        id="new-pet"
        onSubmit={this.onFormSubmit}
        className="dan-class"
        >
        <div>
          <label htmlFor="name">Pet Name</label>
          <input name="name" type="text"
            value={this.state.name}
            onChange={this.onInputChange} />
        </div>
        <div>
          <label htmlFor="breed">Breed</label>
          <input name="breed" type="text"
            value={this.state.breed}
            onChange={this.onInputChange} />
        </div>
        <div>
          <label htmlFor="age">Age</label>
          <input name="age" type="number"
            pattern="[0-9]*"
            value={this.state.age}
            onChange={this.onInputChange} />
        </div>
        <div>
          <label htmlFor="about">About</label>
          <input name="about" type="text"
            value={this.state.about}
            onChange={this.onInputChange} />
        </div>
        <input type="submit" value="Add Pet" />
      </form>
    );
  }
}

export default NewPetForm;
