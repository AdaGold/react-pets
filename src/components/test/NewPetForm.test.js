import React from 'react';
import NewPetForm from '../NewPetForm';
import { shallow } from 'enzyme';

describe('NewPetForm', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow(
      <NewPetForm addPetCallback={() => {} } />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
