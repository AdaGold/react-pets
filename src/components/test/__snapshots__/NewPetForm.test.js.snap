// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm that it matches an existing snapshot 1`] = `
<form
  className="dan-class"
  id="new-pet"
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="name"
    >
      Pet Name
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="breed"
    >
      Breed
    </label>
    <input
      name="breed"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="age"
    >
      Age
    </label>
    <input
      name="age"
      onChange={[Function]}
      pattern="[0-9]*"
      type="number"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="about"
    >
      About
    </label>
    <input
      name="about"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <input
    type="submit"
    value="Add Pet"
  />
</form>
`;
